test/controllers/EditEntryControllerSpec.scala

```diff
@@ -0,0 +1,168 @@
+
+  trait Setup {
+    val defaultSiProtectedUserConfig = siProtectedUserConfigGen.sample.get
+    val defaultAuthStrideEnrolmentsConfigGen = authStrideEnrolmentsConfigGen.sample.get
+    val appName = nonEmptyStringGen.sample.get
+    val stridePid = nonEmptyStringGen.sample.get
+
+    val mockSiProtectedUserListService = mock[SiProtectedUserListService]
+    val mockAuthConnector = mock[AuthConnector]
+    val views = inject[Views]
+    def editEntryController(siProtectedUserConfig: SiProtectedUserConfig = defaultSiProtectedUserConfig) = {
+      new EditEntryController(
+        siProtectedUserConfig,
+        mockSiProtectedUserListService,
+        views,
+        Stubs.stubMessagesControllerComponents(),
+        new StrideAction(mockAuthConnector, defaultAuthStrideEnrolmentsConfigGen, appName)
+      )
+    }
+    def expectStrideAuthenticated(): Unit = {
+      when(mockAuthConnector.authorise[Option[String]](any, any)(any, any)).thenReturn(Future.successful(Some(stridePid)))
+    }
+
+    def assertEditPageContainsFormFields(body: String): Unit = {
+      body should include("edit.entry.title")
+      body should include("edit.entry.header")
+      body should include("entry.form.action")
+      body should include("entry.form.nino")
+      body should include("entry.form.sautr")
+      body should include("entry.form.identityProvider")
+      body should include("entry.form.identityProviderId")
+      body should include("entry.form.addedByTeam")
+      body should include("edit.submit.button")
+      body should include("cancel.button")
+    }
+
+    def toEditRequestFields(entry: Entry): Seq[(String, String)] = {
+      Seq(
+        entry.entryId.map(e => "entryId" -> e),
+        Some("action" -> entry.action),
+        entry.nino.map(n => "nino" -> n),
+        entry.sautr.map(s => "sautr" -> s),
+        entry.identityProvider.map(s => "identityProvider" -> s),
+        entry.identityProviderId.map(s => "identityProviderId" -> s),
+        entry.group.map(s => "group" -> s),
+        entry.addedByTeam.map(s => "addedByTeam" -> s),
+        entry.updatedByTeam.map(s => "updatedByTeam" -> s)
+      ).flatten
+    }
+  }
+
+  "EditEntryController" should {
+    "forward to the edit entry view when GET /add is called" in new Setup {
+      forAll(validEditEntryGen, protectedUserRecordGen) { (entry, protectedUserRecord) =>
+        expectStrideAuthenticated()
+        when(mockSiProtectedUserListService.findEntry(eqTo(entry.entryId.value))(*)).thenReturn(Future.successful(Some(protectedUserRecord)))
+        val result = editEntryController().showEditEntryPage(entry.entryId.value)(FakeRequest().withMethod("GET")).futureValue
+        status(result) shouldBe OK
+
+        val body = contentAsString(result)
+        assertEditPageContainsFormFields(body)
+      }
+    }
+
+    "Forward to confirmation page when edit is successful" in new Setup {
```

Comment:
for PATCH call need to test the scenarios of service throwing an exception (internal server error) and shuttered service

Reply:
As stated above, I'm happy to handle this.

Reply:
resolved